selenium_tests/publish_courses/publish_courses_tests.py

```diff
@@ -0,0 +1,92 @@
+from bulk_utilities.bulk_course_settings import BulkCourseSettingsOperation
+
+from selenium_common.canvas.canvas_course_page_object import CanvasCoursePage
+
+from selenium_tests.publish_courses.publish_courses_base_test_case import \
+    PublishCoursesBaseTestCase
+from selenium_tests.publish_courses.page_objects.main_page import \
+    MainPageObject
+
+
+class PublishCoursesTests(PublishCoursesBaseTestCase):
+
+    def test_publish_courses(self):
+        """TLT-1805: This test case goes through the workflow of publishing
+        all courses in a specific term using the Publish Courses tool """
+
+        main_page = MainPageObject(self.driver)
+
+        #  Select a term from the term dropdown
+        term_with_unpublished_courses = self.term['test_terms'][
+            'term_with_unpublished_courses']
+        main_page.select_term(term_with_unpublished_courses)
+
+        # If the Publish Courses button is enabled, click on it.
+        if main_page.is_publish_all_button_enabled():
+            main_page.publish_courses()
+        else:
+            #  If "Publish all" button is not enabled, unpublish courses
+            #  first to ensure that the publish functionality can be tested.
+            self.bulk_unpublish_canvas_sites()
+
+            #  Refresh the page to see the changes
+            self.refresh_page(term_with_unpublished_courses)
+
+            #  Verify that the Publish Button is clickable
+            self.assertTrue(main_page.is_publish_all_button_enabled())
+            #  Click on "Publish all" button
+            main_page.publish_courses()
+
+        #  Verify that a confirmation appears on screen
+        self.assertTrue(main_page.result_message_visible())
+
+        #  Go to a test course to spotcheck
+        self.driver.get(self.test_course_url)
```

Comment:
This will work if there are very few courses in the term... since it's asynchronous, there's no guarantee that this course will be done by the time we navigate there. In practice though, if there are very few courses, it probably will be! So if it's working consistently, that's good -- just raising it so we're aware of the potential for this to not work as hoped.